mne/preprocessing/tests/test_fine_cal.py

```diff
@@ -126,16 +184,104 @@ def test_compute_fine_cal():
     assert sum([(ic == 1.0).any() for ic in got_cal["imb_cals"]]) == 0
 
     # redoing with 3 imlabance parameters should improve the shielding factor
-    grad_picks = pick_types(raw.info, meg="grad")
-    assert len(grad_picks) == 204 and grad_picks[0] == 0
-    got_grad_imbs = np.array([got_cal["imb_cals"][pick] for pick in grad_picks])
+    grad_subpicks = np.searchsorted(meg_picks, pick_types(raw.info, meg="grad"))
+    assert len(grad_subpicks) == 204 and grad_subpicks[0] in (0, 1)
+    got_grad_imbs = np.array([got_cal["imb_cals"][pick] for pick in grad_subpicks])
     assert got_grad_imbs.shape == (204, 1)
     got_cal_3, _ = compute_fine_calibration(
         raw, cross_talk=ctc, n_imbalance=3, calibration=got_cal, verbose="debug"
     )
-    got_grad_3_imbs = np.array([got_cal_3["imb_cals"][pick] for pick in grad_picks])
+    got_grad_3_imbs = np.array([got_cal_3["imb_cals"][pick] for pick in grad_subpicks])
     assert got_grad_3_imbs.shape == (204, 3)
     corr = np.corrcoef(got_grad_3_imbs[:, 0], got_grad_imbs[:, 0])[0, 1]
-    assert 0.6 < corr < 0.7
+    assert cl3[0] < corr < cl3[1]
     raw_sss_py = maxwell_filter(raw, calibration=got_cal_3, **kwargs)
-    _assert_shielding(raw_sss_py, raw, 68, 70)
+    _assert_shielding(raw_sss_py, raw, *sfs[6:8])
+
+
+@pytest.mark.parametrize(
+    "system",
+    [
+        pytest.param("kit", marks=[pytest.mark.ultraslowtest]),  # ~6s
+        pytest.param("ctf", marks=[td_mark, pytest.mark.ultraslowtest]),  # ~13s
+        pytest.param("fil", marks=[td_mark]),  # ~3s
+        pytest.param("triux", marks=[td_mark, pytest.mark.slowtest]),  # ~7s
+    ],
+)
+def test_fine_cal_systems(system, tmp_path):
+    """Test fine calibration with different systems."""
+    int_order = 8
+    n_ref = 0
+    if system == "kit":
+        raw = read_raw_kit(sqd_path, mrk_path, elp_path, hsp_path)
+        angle_limit = 170
+        err_limit = 500
```

Comment:
err_limit is 500% for kit, and 6000% for CTF?

Reply:
Yeah I think because these are actually not great segments to be processing!